core/sail/api/src/main/java/org/eclipse/rdf4j/sail/helpers/AbstractSailConnection.java

```diff
@@ -459,6 +459,10 @@ public final void rollback()
 						txnPrepared = false;
 					}
 				}
+				else {
+					logger.warn("Cannot rollback transaction on connection because transaction is not active",
+							new Throwable());
```

Comment:
It was in the suggested patch by @elky87. It will generate a stack trace to show where the message came from. Should I wrap it in a system property the same way as we do the other debug stacktraces?